<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>1200</width>
    <height>800</height>
   </size>
  </property>
  <property name="mouseTracking">
   <bool>false</bool>
  </property>
  <property name="windowTitle">
   <string>QR Code Generator &amp; Decoder</string>
  </property>
  <property name="styleSheet">
   <string notr="true">font: 9pt &quot;Segoe UI&quot;;
background-color: rgb(30, 30, 30)</string>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="minimumSize">
    <size>
     <width>1200</width>
     <height>800</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>1200</width>
     <height>800</height>
    </size>
   </property>
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="minimumSize">
       <size>
        <width>1100</width>
        <height>700</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>1100</width>
        <height>700</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="generate">
       <property name="minimumSize">
        <size>
         <width>1100</width>
         <height>700</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>1100</width>
         <height>700</height>
        </size>
       </property>
       <attribute name="icon">
        <iconset theme="list-add"/>
       </attribute>
       <attribute name="title">
        <string>Generate</string>
       </attribute>
       <widget class="QWidget" name="gridLayoutWidget">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>10</y>
          <width>1092</width>
          <height>651</height>
         </rect>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <property name="sizeConstraint">
          <enum>QLayout::SizeConstraint::SetMaximumSize</enum>
         </property>
         <item row="0" column="0">
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <property name="spacing">
            <number>7</number>
           </property>
           <property name="sizeConstraint">
            <enum>QLayout::SizeConstraint::SetMaximumSize</enum>
           </property>
           <property name="leftMargin">
            <number>10</number>
           </property>
           <property name="topMargin">
            <number>10</number>
           </property>
           <property name="rightMargin">
            <number>10</number>
           </property>
           <property name="bottomMargin">
            <number>10</number>
           </property>
           <item>
            <widget class="QGroupBox" name="groupInput">
             <property name="minimumSize">
              <size>
               <width>1070</width>
               <height>20</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>1070</width>
               <height>100</height>
              </size>
             </property>
             <property name="acceptDrops">
              <bool>true</bool>
             </property>
             <property name="title">
              <string>Input</string>
             </property>
             <layout class="QFormLayout" name="formLayout_5">
              <item row="0" column="0">
               <layout class="QFormLayout" name="formLayout">
                <item row="0" column="0" colspan="2">
                 <widget class="QPlainTextEdit" name="insertData">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="minimumSize">
                   <size>
                    <width>1040</width>
                    <height>50</height>
                   </size>
                  </property>
                  <property name="maximumSize">
                   <size>
                    <width>1040</width>
                    <height>100</height>
                   </size>
                  </property>
                  <property name="placeholderText">
                   <string>Enter text / URL / your data here...</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupOptions">
             <property name="maximumSize">
              <size>
               <width>1070</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="title">
              <string>Options</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <layout class="QGridLayout" name="gridLayout_2" columnstretch="3,7">
                <property name="rightMargin">
                 <number>6</number>
                </property>
                <property name="horizontalSpacing">
                 <number>17</number>
                </property>
                <item row="2" column="1">
                 <widget class="QWidget" name="errorCorrectionWidget" native="true">
                  <widget class="QLabel" name="erroCortext">
                   <property name="geometry">
                    <rect>
                     <x>-9</x>
                     <y>1</y>
                     <width>30</width>
                     <height>26</height>
                    </rect>
                   </property>
                   <property name="midLineWidth">
                    <number>2</number>
                   </property>
                   <property name="text">
                    <string/>
                   </property>
                   <property name="alignment">
                    <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
                   </property>
                   <property name="margin">
                    <number>5</number>
                   </property>
                   <property name="indent">
                    <number>20</number>
                   </property>
                  </widget>
                  <widget class="QComboBox" name="errorCorrectionComboBox">
                   <property name="geometry">
                    <rect>
                     <x>2</x>
                     <y>18</y>
                     <width>691</width>
                     <height>24</height>
                    </rect>
                   </property>
                   <item>
                    <property name="text">
                     <string>L (7%)</string>
                    </property>
                   </item>
                   <item>
                    <property name="text">
                     <string>M (15%)</string>
                    </property>
                   </item>
                   <item>
                    <property name="text">
                     <string>Q (25%)</string>
                    </property>
                   </item>
                   <item>
                    <property name="text">
                     <string>H (30%)</string>
                    </property>
                   </item>
                  </widget>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QWidget" name="colorWidget" native="true">
                  <widget class="QWidget" name="horizontalLayoutWidget_2">
                   <property name="geometry">
                    <rect>
                     <x>0</x>
                     <y>0</y>
                     <width>701</width>
                     <height>51</height>
                    </rect>
                   </property>
                   <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="0,2,1,3,2">
                    <item>
                     <widget class="QToolButton" name="btnPickFg">
                      <property name="minimumSize">
                       <size>
                        <width>200</width>
                        <height>40</height>
                       </size>
                      </property>
                      <property name="text">
                       <string>Pick foreground colour</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLabel" name="swatchFg">
                      <property name="minimumSize">
                       <size>
                        <width>28</width>
                        <height>20</height>
                       </size>
                      </property>
                      <property name="maximumSize">
                       <size>
                        <width>45</width>
                        <height>45</height>
                       </size>
                      </property>
                      <property name="frameShape">
                       <enum>QFrame::Shape::Box</enum>
                      </property>
                      <property name="text">
                       <string/>
                      </property>
                      <property name="alignment">
                       <set>Qt::AlignmentFlag::AlignCenter</set>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <spacer name="horizontalSpacer">
                      <property name="orientation">
                       <enum>Qt::Orientation::Horizontal</enum>
                      </property>
                      <property name="sizeHint" stdset="0">
                       <size>
                        <width>15</width>
                        <height>20</height>
                       </size>
                      </property>
                     </spacer>
                    </item>
                    <item>
                     <widget class="QToolButton" name="btnPickBg">
                      <property name="minimumSize">
                       <size>
                        <width>200</width>
                        <height>40</height>
                       </size>
                      </property>
                      <property name="maximumSize">
                       <size>
                        <width>200</width>
                        <height>40</height>
                       </size>
                      </property>
                      <property name="text">
                       <string>Pick background colour</string>
                      </property>
                      <property name="iconSize">
                       <size>
                        <width>40</width>
                        <height>40</height>
                       </size>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLabel" name="swatchBg">
                      <property name="minimumSize">
                       <size>
                        <width>28</width>
                        <height>20</height>
                       </size>
                      </property>
                      <property name="maximumSize">
                       <size>
                        <width>45</width>
                        <height>45</height>
                       </size>
                      </property>
                      <property name="frameShape">
                       <enum>QFrame::Shape::Box</enum>
                      </property>
                      <property name="text">
                       <string/>
                      </property>
                      <property name="alignment">
                       <set>Qt::AlignmentFlag::AlignCenter</set>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </widget>
                 </widget>
                </item>
                <item row="2" column="0">
                 <widget class="QLabel" name="errorCorrectionLabel">
                  <property name="text">
                   <string>Error correction</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignmentFlag::AlignCenter</set>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="colorLabel">
                  <property name="text">
                   <string>Set foreground and background colour</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignmentFlag::AlignCenter</set>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QWidget" name="boxSizeWidget" native="true">
                  <layout class="QGridLayout" name="gridLayout_3">
                   <item row="0" column="0">
                    <widget class="QLabel" name="labelBoxSize">
                     <property name="text">
                      <string>10</string>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="1">
                    <widget class="QSlider" name="sliderBoxSize">
                     <property name="minimum">
                      <number>4</number>
                     </property>
                     <property name="maximum">
                      <number>20</number>
                     </property>
                     <property name="value">
                      <number>10</number>
                     </property>
                     <property name="orientation">
                      <enum>Qt::Orientation::Horizontal</enum>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QLabel" name="boxSizeLabel">
                  <property name="text">
                   <string>Box size</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignmentFlag::AlignCenter</set>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupPreview">
             <property name="maximumSize">
              <size>
               <width>1070</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="title">
              <string>Preview</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
             </property>
             <widget class="QWidget" name="verticalLayoutWidget_3">
              <property name="geometry">
               <rect>
                <x>10</x>
                <y>30</y>
                <width>1052</width>
                <height>201</height>
               </rect>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_6">
               <property name="spacing">
                <number>0</number>
               </property>
               <property name="sizeConstraint">
                <enum>QLayout::SizeConstraint::SetDefaultConstraint</enum>
               </property>
               <item>
                <widget class="QLabel" name="labelQrPreview">
                 <property name="minimumSize">
                  <size>
                   <width>1050</width>
                   <height>100</height>
                  </size>
                 </property>
                 <property name="maximumSize">
                  <size>
                   <width>1040</width>
                   <height>190</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Your preview will show here.</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignmentFlag::AlignCenter</set>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="labelContrastWarning">
                 <property name="maximumSize">
                  <size>
                   <width>1200</width>
                   <height>20</height>
                  </size>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">color: #f1c40f;
font-size: 11px;</string>
                 </property>
                 <property name="text">
                  <string>Low contrast â€“ QR color auto-adjusted</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignmentFlag::AlignCenter</set>
                 </property>
                 <property name="textInteractionFlags">
                  <set>Qt::TextInteractionFlag::NoTextInteraction</set>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnGenerateSave">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>1070</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>1070</width>
               <height>30</height>
              </size>
             </property>
             <property name="layoutDirection">
              <enum>Qt::LayoutDirection::LeftToRight</enum>
             </property>
             <property name="text">
              <string>Generate and Save</string>
             </property>
             <property name="icon">
              <iconset theme="document-save-as"/>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QWidget" name="decode">
       <property name="maximumSize">
        <size>
         <width>1100</width>
         <height>700</height>
        </size>
       </property>
       <attribute name="icon">
        <iconset theme="QIcon::ThemeIcon::DocumentPageSetup"/>
       </attribute>
       <attribute name="title">
        <string>Decode</string>
       </attribute>
       <widget class="QFrame" name="frame">
        <property name="geometry">
         <rect>
          <x>-10</x>
          <y>20</y>
          <width>1200</width>
          <height>901</height>
         </rect>
        </property>
        <property name="frameShape">
         <enum>QFrame::Shape::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Shadow::Raised</enum>
        </property>
        <widget class="QWidget" name="verticalLayoutWidget">
         <property name="geometry">
          <rect>
           <x>60</x>
           <y>40</y>
           <width>931</width>
           <height>561</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <property name="spacing">
             <number>20</number>
            </property>
            <property name="sizeConstraint">
             <enum>QLayout::SizeConstraint::SetDefaultConstraint</enum>
            </property>
            <property name="leftMargin">
             <number>15</number>
            </property>
            <property name="topMargin">
             <number>15</number>
            </property>
            <property name="rightMargin">
             <number>15</number>
            </property>
            <property name="bottomMargin">
             <number>15</number>
            </property>
            <item>
             <widget class="QLabel" name="labelPreview">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="frameShape">
               <enum>QFrame::Shape::StyledPanel</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Shadow::Sunken</enum>
              </property>
              <property name="midLineWidth">
               <number>2</number>
              </property>
              <property name="text">
               <string>Preview</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignmentFlag::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="labelDrop">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="acceptDrops">
               <bool>true</bool>
              </property>
              <property name="styleSheet">
               <string notr="true">background-color: rgb(182, 182, 182)</string>
              </property>
              <property name="frameShape">
               <enum>QFrame::Shape::StyledPanel</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Shadow::Sunken</enum>
              </property>
              <property name="midLineWidth">
               <number>2</number>
              </property>
              <property name="text">
               <string>Drop file / Click to add file from disk</string>
              </property>
              <property name="textFormat">
               <enum>Qt::TextFormat::AutoText</enum>
              </property>
              <property name="alignment">
               <set>Qt::AlignmentFlag::AlignCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="labelResults">
            <property name="text">
             <string>Results</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignmentFlag::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>1000</width>
              <height>5</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QTextEdit" name="showResultsLabel">
            <property name="focusPolicy">
             <enum>Qt::FocusPolicy::NoFocus</enum>
            </property>
            <property name="frameShape">
             <enum>QFrame::Shape::Box</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Shadow::Plain</enum>
            </property>
            <property name="textInteractionFlags">
             <set>Qt::TextInteractionFlag::TextSelectableByKeyboard|Qt::TextInteractionFlag::TextSelectableByMouse</set>
            </property>
            <property name="placeholderText">
             <string>Your results will show here.</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>10</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="BtnCopyResults">
            <property name="text">
             <string>Copy results</string>
            </property>
            <property name="icon">
             <iconset theme="QIcon::ThemeIcon::EditCopy"/>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>10</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <action name="actionGenerate">
   <property name="text">
    <string>Generate</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>sliderBoxSize</sender>
   <signal>valueChanged(int)</signal>
   <receiver>labelBoxSize</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>738</x>
     <y>210</y>
    </hint>
    <hint type="destinationlabel">
     <x>583</x>
     <y>207</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
